### DISCUSSION TITLE
Can't find TransformStream in some iOS devices

### DISCUSSION THREAD

**Discussion by FishHawk** [gildas-lormeau/zip.js#489] [Q&A] [answered]
Some iOS users of my website reported that `zip.js` is causing an error: `ReferenceError: Can't find variable: TransformStream`.  Even after updating or changing browsers, the problem remains. However, using the `web-streams-polyfill` resolves the issue.

Could this just be a browser compatibility problem? If so, why doesn't updating the browser fix it?

**Comment by gildas-lormeau** [comment, marked as answer]
I confirm that `zip.js` heavily depends on Web Streams (i.e. `WritableStream`, `ReadableStream` and `TransformStream`). It's pretty weird that updating browsers doesn't solve the problem. `TransformStream` is supposed to be supported on Safari since the version 14.1 released on 26 Apr. 2021, see https://caniuse.com/mdn-api_transformstream_transformstream.

**Reply by FishHawk** [reply]
Thanks for your answer.

After checking again, the users have not updated their iOS. Even if the browser version is updated, it remains very old. However, according to "caniuse", these browsers are still supported.

The browser versions:
- Chrome 93.0.4577.78
- Safari 14.2

**Reply by FishHawk** [reply]
It's still strange, but I think it's just a compatibility issue.
